### PR TITLE
Authentication with auth0

### PR DESCRIPTION
**Pull request by alan-maldonado**
# Overview
This PR contains the necessary functionality to log in with auth0.

# Configuration
We have to copy `.env.sample` to `.env` and set up auth0_client_id and auth_domain. I can share my development keys if needed.

# Demo:
http://g.recordit.co/ecQN125mND.gif

# Jira Issue
https://wizeline.atlassian.net/browse/WIP-198

## Considerations.
- Login component is without a parent.
- Profile and auth.js are from boilerplate https://github.com/auth0-community/auth0-react-scripts
- Once we define structured mockups, we can move these components.
- No linter analysis has been applied

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+const NEXT_PATH_KEY = 'next_path';
+const ID_TOKEN_KEY = 'id_token';
+const ACCESS_TOKEN_KEY = 'access_token';
+const PROFILE_KEY = 'profile';
+const LOGIN_ROUTE = '/login';
+const ROOT_ROUTE = '/';
+
+if (!process.env.REACT_APP_AUTH0_CLIENT_ID || !process.env.REACT_APP_AUTH0_DOMAIN) {
+  throw new Error('Please define `REACT_APP_AUTH0_CLIENT_ID` and `REACT_APP_AUTH0_DOMAIN` in your .env file');
+}
```
Comment (codinronan) [inline src/auth.js]: Oh well, since we are using create-react-app that is hard. Ok ummm I'll look into it.

File: src/auth.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  return {
+    close() {
+      events.removeListener('profile_updated', subscription);
+    }
+  };
+}
+
+async function updateProfile(userId, newProfile) {
+  try {
+    const response = await fetchAsUser(`https://${process.env.REACT_APP_AUTH0_DOMAIN}/api/v2/users/${userId}`, {
```
Comment (codinronan) [inline src/auth.js]: Wow, CRA now includes async/await. Awesome.

File: src/components/EditProfile.css [hunk 3]
```diff
@@ -0,0 +1,49 @@
+.EditProfile {
```
Comment (alan-maldonado) [inline src/components/EditProfile.css]: I really don't like this also, but I didn't want to change the boilerplate project. 
I can do.

File: src/auth.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    const profile = await response.json();
+    setProfile(profile);
+  } catch (error) {
+    return error;
+  }
+}
+
+function setProfile(profile) {
+  localStorage.setItem(PROFILE_KEY, JSON.stringify(profile));
+  events.emit('profile_updated', profile);
```
Comment (alan-maldonado) [inline src/auth.js]: yeah, agree with you. Do you want me to do some setup for redux, or wait for a new hire to explore redux?

Comment (codinronan) [inline src/auth.js]: Nope that's fine, actually thinking about it now this is probably too simple an app for redux. If not, it's easy enough to add in later. It was just an observation :)

File: src/components/EditProfile.css (hunk 3 above)
Comment (codinronan) [inline src/components/EditProfile.css]: Nah that's ok.

**Comment by codinronan** [comment]
:shipit:

File: src/components/EditProfile.css (hunk 3 above)
Comment (rominavarela) [inline src/components/EditProfile.css]: I simply vote for keeping which ever convention that exists in the project. If there is none, I prefer snake-case for web.

**Pull request merged**